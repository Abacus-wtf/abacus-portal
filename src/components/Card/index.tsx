import React from "react"
import styled from "styled-components"
import Countdown from "react-countdown"
import { theme } from "@config/theme"
import { User } from "react-feather"
import { StaticImage } from "gatsby-plugin-image"
import { Text, ImageContainer } from "@components/global.styles"
import { SessionData } from "@state/sessionData/reducer"

const CardContainer = styled.div`
  width: 100%;
  height: 100%;
  min-height: 350px;
  max-width: 400px;
  display: flex;
  flex-direction: column;
  align-items: center;
  grid-gap: 14px;
  transition: 0.2s;
  opacity: 1;
  overflow: hidden;

  &:hover {
    opacity: 0.6;
  }
`

const MiniText = styled.div`
  background: rgba(0, 0, 0, 0.65);
  max-height: min-content;
  border-radius: 35px;
  padding: 4px 10px;
  max-height: 30px;
  font-size: 14px;
  backdrop-filter: blur(10px);
  color: ${theme.colors.text3};
`

const UserStyled = styled(User)`
  height: 13px;
  width: 16px;
  margin-top: -2px;
`

const OuterTextContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  grid-gap: 4px;
`

const TextContainer = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: flex-end;
  grid-gap: 20px;
`

const BoldText = styled(Text)`
  font-weight: bold;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: 60%;
  overflow: hidden;
`

const SubText = styled(Text)`
  color: ${theme.colors.text2};
  font-size: 0.85rem;
  text-overflow: ellipsis;
  white-space: nowrap;
`

const EthText = styled(BoldText)`
  font-size: 1.2rem;
  display: flex;
  flex-direction: row;
  grid-gap: 5px;
  align-items: center;
  margin-bottom: -1px;
`

type CardProps = {
  tokenId: SessionData["tokenId"]
  image_url: SessionData["image_url"]
  endTime: SessionData["endTime"]
  numPpl: SessionData["numPpl"]
  nftName: SessionData["nftName"]
  finalAppraisalValue?: SessionData["finalAppraisalValue"]
  totalStaked: SessionData["totalStaked"]
  collectionTitle: SessionData["collectionTitle"]
}

export default ({
  tokenId,
  image_url,
  endTime,
  numPpl,
  nftName,
  finalAppraisalValue,
  totalStaked,
  collectionTitle,
}: CardProps) => (
  <CardContainer>
    <ImageContainer src={image_url}>
      <MiniText>
        <Countdown
          date={endTime * 1000}
          renderer={({ hours, minutes, seconds, completed }) => {
            if (completed) {
              return <>Completed</>
            }
            return (
              <span>
                {hours}:{minutes}:{seconds}
              </span>
            )
          }}
        />
      </MiniText>
      <MiniText>
        <UserStyled /> {numPpl}
      </MiniText>
    </ImageContainer>
    <OuterTextContainer>
      <TextContainer>
        <BoldText>
          {nftName} #
          {tokenId.length >= 8 ? `${tokenId.slice(0, 8)}...` : tokenId}
        </BoldText>
        <EthText>
          <StaticImage
            alt=""
            style={{ height: 15 }}
            src="../../images/ETH.svg"
          />{" "}
          {finalAppraisalValue !== undefined
            ? finalAppraisalValue.toLocaleString("en-US", {
                minimumFractionDigits: 2,
                maximumFractionDigits: 4,
              })
            : totalStaked.toLocaleString("en-US", {
                minimumFractionDigits: 2,
                maximumFractionDigits: 4,
              })}
        </EthText>
      </TextContainer>
      <TextContainer>
        <SubText style={{ maxWidth: "min-content", overflow: "hidden" }}>
          {collectionTitle}
        </SubText>
        <SubText data-testid="subtext">
          {finalAppraisalValue !== undefined
            ? "Final Appraisal"
            : "Total Staked"}
        </SubText>
      </TextContainer>
    </OuterTextContainer>
  </CardContainer>
)
