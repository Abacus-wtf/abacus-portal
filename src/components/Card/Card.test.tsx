import { getByTestId, render } from "@test-utils/index"
import React from "react"
import "@testing-library/jest-dom/extend-expect"
import Card from "./index"

const defaultProps = {
  tokenId: "5007",
  image_url: "http://example.com",
  endTime: 0,
  numPpl: 2,
  nftName: "Deez Nuts",
  finalAppraisalValue: 22,
  totalStaked: 1,
  collectionTitle: "Lazy Bird",
}

describe("Card", () => {
  it("Matches snapshot", () => {
    const { container } = render(<Card {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it("shows Total Staked when no finalAppraisalValue", () => {
    const { container } = render(
      <Card {...defaultProps} finalAppraisalValue={undefined} />
    )
    const { innerHTML } = getByTestId(container, "subtext")
    expect(innerHTML).toBe("Total Staked")
  })
})
