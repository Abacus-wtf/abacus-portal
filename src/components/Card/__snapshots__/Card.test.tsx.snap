// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Matches snapshot 1`] = `
<div>
  <div
    class="sc-jrAFXE gKtaCv"
  >
    <div
      class="sc-fubCzh FBFtj"
      src="http://example.com"
    >
      <div
        class="sc-kEjbQP dfbwfv"
      />
      <div
        class="sc-kEjbQP dfbwfv"
      >
        <svg
          class="sc-iqHYmW iRfQUM"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
          />
          <circle
            cx="12"
            cy="7"
            r="4"
          />
        </svg>
         
        2
      </div>
    </div>
    <div
      class="sc-crrszt iNxCEk"
    >
      <div
        class="sc-dQoVA gdUAbl"
      >
        <span
          class="sc-bdfBQB sc-bqyKOL jExror hwLekN"
        >
          Deez Nuts
           #
          5007
        </span>
        <span
          class="sc-bdfBQB sc-bqyKOL sc-hBEYId jExror hwLekN fybUBL"
        >
          <img
            alt=""
            src=""
            style="height: 15px;"
          />
           
          22.00
        </span>
      </div>
      <div
        class="sc-dQoVA gdUAbl"
      >
        <span
          class="sc-bdfBQB sc-kstqJO jExror ifOUGr"
          style="max-width: min-content; overflow: hidden;"
        >
          Lazy Bird
        </span>
        <span
          class="sc-bdfBQB sc-kstqJO jExror ifOUGr"
          data-testid="subtext"
        >
          Final Appraisal
        </span>
      </div>
    </div>
  </div>
</div>
`;
